Azure/iotedge
# [k8s] Make EdgeDeployment set so Edge Agent does not update for public preview

Extract image out of current agent image, and reuse it for edge
deployment definition.  That way, edgeAgent is only launched with
the image provided to iotedged.

Created a unit test for this.

To test E2E:
Checkout branch (or master once merged).
Use edge agent image built with this change.
Create a default Edge Deployment on an edge device in IoT Hub.

Deploy edge-on-k8s from github repo:
```sh
cd <repo>/kubernetes/charts
helm install ./edge-kubernetes-crd
helm install \
  --name edgy \
  --namespace edgy-iot-1 \
  --set "deviceConnectionString=HostName=<device connection string>" \
  --set "edgeAgent.image.repository=<agent build image>"
  --set "edgeAgent.image.tag=<agent-build-tag>"
  -f override.yaml \
  ./edge-kubernetes
```

Let the edge agent start.  EdgeAgent pod should come up and remain running, no restarts.  
If you go to the Edge Device page, $edgeAgent desired module image should be `mcr.microsoft.com/azureiotedge-agent:1.0` and the reported module image should be: `<agent build image>:<agent-build-tag>`

test passed E2E.
